File: python/palindrome_number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        origin_num = x
        rev = 0
        while x > 0:
            rev = rev * 10 + x % 10
            x //= 10
        return origin_num == rev

#test
test = 121
result = Solution()
print(f"{test} {result.isPalindrome(test)}")

File: rust/palindrome_number.rs
    pub fn is_palindrome(x: i32) -> bool{
        if x < 0 {
            return false;
        }
        let mut x = x;
        let mut rev = 0;

        while x > rev {
            rev = rev * 10 + x % 10;
            x /= 10;
        }
        x == rev || x == rev / 10
    }

fn main() {
let test = 121;
let result = is_palindrome(test);
println!("{:?}", result);
}
